packages/hoppscotch-app/src/helpers/teams/TeamEnvironmentAdapter.ts

```diff
@@ -0,0 +1,241 @@
+    this.timeoutHandle = null
+    this.isDispose = false
+
+    this.teamEnvironmentCreated$ = null
+    this.teamEnvironmentDeleted$ = null
+    this.teamEnvironmentUpdated$ = null
+    this.teamEnvironmentCreatedSub = null
+    this.teamEnvironmentDeletedSub = null
+    this.teamEnvironmentUpdatedSub = null
+
+    if (teamID) this.initialize()
+  }
+
+  unsubscribeSubscriptions() {
+    this.teamEnvironmentCreated$?.unsubscribe()
+    this.teamEnvironmentDeleted$?.unsubscribe()
+    this.teamEnvironmentUpdated$?.unsubscribe()
+    this.teamEnvironmentCreatedSub?.unsubscribe()
+    this.teamEnvironmentDeletedSub?.unsubscribe()
+    this.teamEnvironmentUpdatedSub?.unsubscribe()
+  }
+
+  changeTeamID(newTeamID: string | undefined) {
+    this.teamID = newTeamID
+    this.teamEnvironmentList$.next([])
+    this.loading$.next(false)
+
+    this.unsubscribeSubscriptions()
+
+    if (this.teamID) this.initialize()
+  }
+
+  async initialize() {
+    if (this.timeoutHandle) throw new Error(`Adapter already initialized`)
+    if (this.isDispose) throw new Error(`Adapter has been disposed`)
+
+    await this.fetchList()
+    this.registerSubscriptions()
+  }
+
+  public dispose() {
+    this.isDispose = true
+    clearTimeout(this.timeoutHandle as any)
+    this.timeoutHandle = null
+    this.unsubscribeSubscriptions()
+  }
+
+  async fetchList() {
+    if (this.teamID === undefined) throw new Error("Team ID is null")
+
+    this.loading$.next(true)
+
+    const results: TeamEnvironment[] = []
+
+    const result = await runGQLQuery({
+      query: GetTeamEnvironmentsDocument,
+      variables: {
+        teamID: this.teamID,
+      },
+    })
+
+    if (E.isLeft(result)) {
+      this.error$.next(result.left)
+      this.loading$.next(false)
+      console.error(result.left)
+      throw new Error(`Failed fetching team environments: ${result.left}`)
+    }
+
+    if (result.right.team !== undefined && result.right.team !== null) {
+      results.push(
+        ...result.right.team.teamEnvironments.map(
+          (x) =>
+            <TeamEnvironment>{
+              id: x.id,
+              teamID: x.teamID,
+              environment: {
+                name: x.name,
+                variables: JSON.parse(x.variables),
+              },
+            }
+        )
+      )
+    }
+
+    this.teamEnvironmentList$.next(results)
+
+    this.loading$.next(false)
+  }
+
+  private createNewTeamEnvironment(newEnvironment: TeamEnvironment) {
+    const teamEnvironments = this.teamEnvironmentList$.value
+
+    teamEnvironments.push(newEnvironment)
+
+    this.teamEnvironmentList$.next(teamEnvironments)
+  }
+
+  private deleteTeamEnvironment(envId: string) {
+    const teamEnvironments = this.teamEnvironmentList$.value.filter(
+      ({ id }) => id !== envId
+    )
+
+    this.teamEnvironmentList$.next(teamEnvironments)
+  }
+
+  private updateTeamEnvironment(updatedEnvironment: TeamEnvironment) {
```

Comment:
Team Environment renames do not trigger updates.

Steps to repro:
1. Use Hoppscotch with 2 users (single user also has issue, but this should work too)
2. Point both of them to the same team and use Team Environments.
3. Create a Team Environment.
4. Then, rename the team environment.
5. Observe that the environment has not been renamed.

I haven't checked if its just the name or other parts too, please do check.